### PR TITLE
goaws : test fake sqs/sns server

### PR DESCRIPTION
**Pull request by deeptibaghel**
PR to test goaws : fake sqs/sns server

### CODE REVIEW

File: lib/notifier.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ if (notifierSnsTopicArn !== 'disabled') {
   // This will pull in default credentials, region data etc
   // from the metadata service available to the instance.
   // It's magic, and it's awesome.
-  sns = new AWS.SNS({region: region})
+  sns = new AWS.SNS({endpoint: 'http://localhost:4100/', region: region})
```
Comment (vladikoff) [inline lib/notifier.js]: @deeptibaghel to merge this let's make the `endpoint` configurable. 

Add a new block into https://github.com/mozilla/fxa-auth-server/blob/master/config/index.js#L442 before `snsTopicArn`, we can call this `snsTopicEndpoint` for now. default will be `undefined`. and `http://localhost:4100/` will be set in `dev.json`

Comment (vladikoff) [reply lib/notifier.js]: sorry `snsTopicEndpoint` will be listed after `snsTopicArn` in the config

**Comment by vladikoff** [comment]
@deeptibaghel plus one more thing , let's add a guard to `config/index.js` such that if `env` is `production` then the `endpoint` **must NOT** be set. Example of similar code: https://github.com/mozilla/fxa-content-server/blob/master/server/lib/configuration.js#L642-L644

(Basically this means you can only change the `endpoint` setting in `development` mode

**Comment by vladikoff** [comment]
@deeptibaghel one more thing, we have to set 

`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` if they are NOT set as env variable. to handle the case where aws-sdk throws an error

**Comment by deeptibaghel** [comment]
> AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY if they are NOT set as env variable. to handle the case where aws-sdk throws an error

@vladikoff , we need to set it in which file ?

File: config/index.js [hunk 2]
```diff
@@ -838,4 +844,8 @@ conf.set('smtp.verifySecondaryEmailUrl', conf.get('contentServer.url') + '/verif
 
 conf.set('isProduction', conf.get('env') === 'prod')
 
+if (conf.get('env') === 'prod') {
+  conf.set('snsTopicEndpoint', undefined);
```
Comment (deeptibaghel) [inline config/index.js]: @vladikoff , is that ok or we need to "delete" it.

Comment (vladikoff) [reply config/index.js]: @deeptibaghel let's do this check: `IF snsTopicEndpoint is set AND env is NOT dev THEN throw new Error('snsTopicEndpoint is only allowed in dev env')`

Comment (vladikoff) [inline config/index.js]: also remove the `;`

Comment (deeptibaghel) [reply config/index.js]: This is done, now i think the credentials are needed.

File: config/index.js [hunk 3]
```diff
@@ -845,8 +845,17 @@ conf.set('smtp.verifySecondaryEmailUrl', conf.get('contentServer.url') + '/verif
...
 
 //sns endpoint is not to be set in production
-if (conf.has('snsTopicEndpoint') && conf.get('env') !== 'dev' ) {
+if (conf.has('snsTopicEndpoint') && conf.get('env') !== 'dev') {
   throw new Error('snsTopicEndpoint is only allowed in dev env')
 }
 
+if(conf.get('env')  === 'dev'){
+  if (! process.env.AWS_ACCESS_KEY_ID) {
+    process.env.AWS_ACCESS_KEY_ID = 'foo'
```
Comment (vladikoff) [inline config/index.js]: let's rename `foo` to `DEV_KEY_ID` and the `bar` below to `DEV_ACCESS_KEY`. if we ever see this in the process env if will make more sense where these values came from.

File: config/index.js [hunk 4]
```diff
@@ -838,4 +844,18 @@ conf.set('smtp.verifySecondaryEmailUrl', conf.get('contentServer.url') + '/verif
 
 conf.set('isProduction', conf.get('env') === 'prod')
 
+//sns endpoint is not to be set in production
+if (conf.has('snsTopicEndpoint') && conf.get('env') !== 'dev') {
+  throw new Error('snsTopicEndpoint is only allowed in dev env')
+}
+
+if(conf.get('env')  === 'dev'){
```
Comment (vladikoff) [inline config/index.js]:
```
/home/travis/build/mozilla/fxa-auth-server/config/index.js
  852:1  error  Expected space(s) after "if"  keyword-spacing
```

File: config/index.js [hunk 5]
```diff
@@ -838,4 +844,18 @@ conf.set('smtp.verifySecondaryEmailUrl', conf.get('contentServer.url') + '/verif
 
 conf.set('isProduction', conf.get('env') === 'prod')
 
+//sns endpoint is not to be set in production
+if (conf.has('snsTopicEndpoint') && conf.get('env') !== 'dev') {
+  throw new Error('snsTopicEndpoint is only allowed in dev env')
+}
+
+if (conf.get('env')  === 'dev'){
```
Comment (vladikoff) [inline config/index.js]: @deeptibaghel there are 2 spaces before ` ===`

**Pull request merged**